pymeasure/instruments/keysight/keysight81160A.py

```diff
@@ -0,0 +1,208 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.agilent import Agilent33500
+from pymeasure.instruments.agilent.agilent33500 import Agilent33500Channel
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+WF_SHAPES = ["SIN", "SQU", "RAMP", "PULS", "NOIS", "DC", "USER"]
+FREQUENCY_RANGE = [1e-6, 5e8]
+AMPLITUDE_RANGE = [0.05, 5]
+OFFSET_RANGE = [-4.995, 4.995]
+VOLTAGE_HIGH_RANGE = [-4.95, 5.0]
+VOLTAGE_LOW_RANGE = [-5, 4.95]
+SQUARE_DUTYCYCLE_RANGE = [0.0001515, 99.9998485]
+PULSE_PERIOD_RANGE = [2e-9, 1e6]
+PULSE_WIDTH_RANGE = [1.5e-9, 1e6]
+PULSE_DUTYCYCLE_RANGE = [1.5e-1, 99.99]
+PULSE_TRANSITION_RANGE = [1e-9, 1e3]
+BURST_PERIOD = [3.03e-9, 1e6]
+BURST_NCYCLES = [2, 2.147483647e9]
+LIMIT_HIGH_RANGE = [-9.99, 10]
+LIMIT_LOW_RANGE = [-10, 9.99]
+STATES = ["ON", "OFF"]
+IMPEDANCE_RANGE = [3e-1, 1e6]
+
+
+class Keysight81160AChannel(Agilent33500Channel):
+    shape_values = WF_SHAPES
+    shape_get_command = ":FUNC{ch}?"
+    shape_set_command = ":FUNC{ch} %s"
```

Comment:
I'm unsure, whether it is better to write the properties anew or use their dynamic parts (as you do here).
In this case, you basically just copy the docstring and validator from the inherited class.